zerver/models.py

```diff
@@ -1094,6 +1096,23 @@ def is_claimed(self):
     def get_url(self):
         return "/user_uploads/%s" % (self.path_id)
 
+
+def validate_attachment_request(user_profile, path_id):
+    try:
+        attachment = Attachment.objects.get(path_id=path_id)
+        messages = attachment.messages.all()
```

Comment:
@timabbott I tried to do the query in one shot but I couldn't. `UserMessage.objects.filter(user_profile=user_profile, message__in=attachment.messages).exists()` throws an error and the stack trace wasn't helpful either. I think its because there isn't a direct relation between UserMessage and Attachment or because messages field is ManyToMany.

We can try caching `messages = attachment.messages.all()` to improve the performance :)